woocommerce/woocommerce-gutenberg-products-block
# Add storybook story for icon library

This pull adds a storybook story for the `@woocommerce/icons` library.

<img width="1380" alt="Image 2020-06-25 at 5 36 45 PM" src="https://user-images.githubusercontent.com/1429108/85797874-75ac3f00-b70a-11ea-8744-59c856e0deda.png">

This is pretty much a direct take from the equivalent Gutenberg implementation except I modified the styling a bit so more icons display on the screen. The story is setup so that any new icons added to the library will automatically display with the story.

In the process of doing this pull the story revealed the following:

- There was direct use of `wp` global for filters we were using instead of importing from the package (external)
- The dashIcon replacements we were injecting in lieu of the full dashicon library were being exported as react elements instead of nodes from the icon library which was inconsistent with other icons in the library. I switched that to export node, and then updated the dashicon module replacement to use `cloneElement` instead of `createElement`.
- The woo icon used for our block category was also being exported as a react element instead of node. So I fixed that too. It also needed updated to correctly receive `width`, `height` and other props (so cloning to pass those values through would work). I updated the implementation where the category is registered so that it's implemented using the `Icon` component pattern used for all icons in the library.
- The `arrowBack` icon from the library didn't have `viewBox` set, so that meant it wasn't reflecting changes in size properly.

## To Test

The story can be tested by running `npm run storybook` and viewing the story for the icons. When this pull is merged to master they should show up in the generated storybook as well.

Due to the changes in other files, there should be some smoke testing of the impacted code:

### wp global to package import

So the interesting thing here is that this code is related to the work don in [this pull](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1613). However, I discovered while testing, that although the grid blocks are initializing with the expected defaults, all attributes _are not_ getting saved with the blocks as expected. This is also happening on master too. So I'm not sure if there was another change that impacted this (maybe filters loading too late?) but if this behaviour is unexpected we'll likely need to investigate via a follow-up issue (cc @mikejolley).

### Dashicon replacements.

The affected icons are the chevron on our selects (in checkout):

<img width="320" alt="Image 2020-06-25 at 5 11 26 PM" src="https://user-images.githubusercontent.com/1429108/85798300-3d593080-b70b-11ea-9880-a5c9f2db480a.png">

...and the "x" dismiss icon in notifications:

<img width="1157" alt="Image 2020-06-25 at 5 50 26 PM" src="https://user-images.githubusercontent.com/1429108/85799004-85c51e00-b70c-11ea-9275-2992717e6c25.png">

* [ ] Verify that they icons appear and there are no errors in the console in both the frontend and backend.

### Woo Icon

Verify that the woo icon displays correctly for the WooCommerce Blocks category in the Block picker.

### `arrowBack` icon.

This icon is used with the "Return to Cart" item in the checkout block:

<img width="281" alt="Image 2020-06-25 at 5 13 32 PM" src="https://user-images.githubusercontent.com/1429108/85798463-83ae8f80-b70b-11ea-802c-08270f97c1bf.png">

* [ ] Verify that the `arrowBack` icon displays as expected.
